packages/next/src/build/webpack/plugins/flight-client-entry-plugin.ts

```diff
@@ -177,6 +180,25 @@ export class ClientReferenceEntryPlugin {
           webpack.dependencies.ModuleDependency,
           new webpack.dependencies.NullDependency.Template()
         )
+
+        // TODO: This might be worth extracting to a separate plugin in the future
+        // if it becomes a more common pattern. We keep it here for now to avoid
+        // extra overhead of a new plugin.
+        normalModuleFactory.hooks.module.tap(PLUGIN_NAME, (module) => {
```

Comment:
You should be able to use this in package.json in next:

```js
"sideEffects": [
  "dist/esm/server/web/exports/image-response.js",
  ...
]
```

Reply:
Hmm... Actually you need the inverse... Might be more involved.

module.rules also allows to set side effects: false, that is probably easier

Reply:
I went through issues in the Webpack repo and what we really want is:

```js
("@pure");
import xyz from "xyz";
```